main.tf

```diff
@@ -15,16 +15,21 @@
  */
 
 terraform {
-  required_version = ">= 0.12"
+  # 0.12.6 introduced for_each for resources
+  required_version = ">= 0.12.6"
 }
 
 locals {
   prefix = var.prefix == "" ? "" : "${var.prefix}-"
+  names_map = zipmap(var.names, var.names)
+  folders = values(google_folder.folders)
+  first_folder = local.folders[0]
```

Comment:
This should be the folder corresponding to the first name in `var.names`, not the first folder in lexical order which is what you're setting here.

Reply:
fixed in 5fe55f9